fix(common): allow null in ngComponentOutlet

crisbeto opened the pull request:
`ngComponentOutlet` already handles null/undefined values, but the types don't reflect that. These changes update the types.

Fixes #45716.

crisbeto commented on packages/common/src/directives/ng_component_outlet.ts:
```diff
@@ -70,7 +70,7 @@ import {ComponentRef, createNgModuleRef, Directive, Injector, Input, NgModuleFac
 @Directive({selector: '[ngComponentOutlet]'})
 export class NgComponentOutlet implements OnChanges, OnDestroy {
-  @Input() ngComponentOutlet!: Type<any>;
+  @Input() ngComponentOutlet?: Type<any>|null = null;
```
Not a fan of marking both as optional and null, but I wanted to keep it consistent with the other inputs.

AndrewKushnir replied:
The `ngComponentOutlet` input corresponds to the directive selector, so it should always be set and we can drop the `?`?

crisbeto replied:
Done

crisbeto commented on packages/common/src/directives/ng_component_outlet.ts:
```diff
@@ -81,7 +81,6 @@ export class NgComponentOutlet implements OnChanges, OnDestroy {
    */
   @Input() ngComponentOutletNgModuleFactory?: NgModuleFactory<any>;
 
-  private _componentRef: ComponentRef<any>|undefined;
```
I removed this since we were assigning to it, but we weren't reading it anywhere.

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

crisbeto committed:
fix(common): allow null in ngComponentOutlet

`ngComponentOutlet` already handles null/undefined values, but the types don't reflect that. These changes update the types.

Fixes #45716.

AndrewKushnir reviewed (approved):
Thanks for the update @crisbeto 👍

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

atscott reviewed (approved):
reviewed-for: public-api

jessicajaniuk reviewed (approved):
reviewed-for: public-api
